### ISSUE TITLE
Resolve Fenix breaking changes for sqlcipher remove

### ISSUE THREAD

**Issue by bendk** [mozilla/application-services#4443]
This an issue to handle the Fenix side of  #4200.  We should resolve the changes created by the sqlcipher logins migration in the Fenix  repo. This will include the following:

- Replacing logins constructors with the per-field encryption constructors
- Replacing logins `add` and `update` calls with the new `add_or_update` function
  - **Note:** This function will have an encryption key parameter.
- Removing references to `check_valid_with_no_dupes` and `potential_dupes_ignoring_username` functions
  - **Note:** This update may require a-c/fenix resources as [some of the views](https://github.com/mozilla-mobile/android-components/blob/1a9b3eaf58a8a73134926a2ba98141600ae422d3/components/feature/prompts/src/main/java/mozilla/components/feature/prompts/dialog/SaveLoginDialogFragment.kt#L334) _may_ need to change depending how the logic is rewritten
- Adding `decrypt_and_fixup_login` calls before all login `get_all`, `get_by_base_domain`, and `get_by_id` calls
  - **Note:** This function will have an encryption key parameter.
- Adding an encryption key parameter to `import_multiple` calls  
- Fixing all broken tests and adding tests where sensible

Feel free to break this work out further where it makes sense. For instance, replacing `add` and `update` and removing `check_valid_with_no_dupes` and `potential_dupes_ignoring_username` could be split into a separate ticket. However, performing this work in parallel might be challenging.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/SYNC-2777)
┆Story Points: 3
┆Epic: Logins update and sqlcipher migration
┆Fix Versions: Release 95
┆Sprint End Date: 2021-10-15
